### PR TITLE
Accept namespaces, learning rate range, regularization range, as input; started work towards automating latter two

### PR DESCRIPTION
**Pull request by sidsen**

### CODE REVIEW

**Comment by marco-rossi29** [comment]
Commit 1096e39 is already in master from pull request 112

**Comment by marco-rossi29** [comment]
@sidsen Do you want to merge this pull request or are you still working on it?

**Comment by sidsen** [comment]
This is ready to be merged. I will continue working on automation but this
is an intermediate step that works.

Sid

File: DataScience/Experimentation.py [hunk 1]
```diff
@@ -119,70 +120,103 @@ def detect_namespaces(j_obj, ns_set, marginal_set=None):
...
     parser = argparse.ArgumentParser()
-    parser.add_argument('-f','--file_path', help="data file path", required=True)
-    parser.add_argument('-q','--max_q_terms', type=int, help="number of quadratic terms to explore with brute-force", default=3)
-    parser.add_argument('-p','--n_proc', type=int, help="number of parallel processors used", default=multiprocessing.cpu_count()-1)
-    parser.add_argument('-b','--base_command', help="base command before data file path (default: vw --cb_adf --dsjson -c -d )", default='vw --cb_adf --dsjson -c -d ')
+    parser.add_argument('-f','--file_path', help="data file", required=True)
+    parser.add_argument('-q','--max_q_terms', type=int, help="number of quadratic terms to explore with brute-force (default: 2)", default=2)
+    parser.add_argument('-p','--n_proc', type=int, help="number of parallel processes to use (default: auto-detect)", default=multiprocessing.cpu_count()-1)
+    # Construct default VW path in platform-agnostic away
+    vwPath = os.path.join('.', 'vw')
```
Comment (marco-rossi29) [inline DataScience/Experimentation.py]: This does not work in windows.

.\vw --cb_adf --dsjson -c -d C:\ds_workshop\ds_data_tag.json -l 0.5

FileNotFoundError: [WinError 2] The system cannot find the file specified

vw is not found

File: DataScience/Experimentation.py [hunk 2]
```diff
@@ -305,12 +339,13 @@ def detect_namespaces(j_obj, ns_set, marginal_set=None):
     best_learning_rate = results[0].learning_rate
 
     # TODO: Repeat above process of tuning parameters and interactions until convergence / no more improvements.
-    
-    print("Best parameters:")
+
+    elapsed_time = datetime.now() - t0
+    elapsed_time -= timedelta(microseconds=elapsed_time.microseconds)
```
Comment (marco-rossi29) [inline DataScience/Experimentation.py]: What is the purpose of this? Just removing microseconds?

Do:
`print("\nBest parameters found after elapsed time {0}:".format(str(elapsed_time).split('.')[0])`

File: DataScience/Experimentation.py (hunk 1 above)
Comment (sidsen) [reply DataScience/Experimentation.py]: I'll test this gain on linux and cygwin on Windows -- is there a non-cygwin way you are running vw's executable on Windows? Also, I just realized we don't currently allow the vw path to be specified via the CLI -- I'll add support for that.

File: DataScience/Experimentation.py (hunk 2 above)
Comment (sidsen) [reply DataScience/Experimentation.py]: Yes, the microsecond output was too long and also not useful (experimentation takes long enough that second granularity is enough). I'll add a comment to reflect this, unless you have a suggestion for doing this in a more readable way.

File: DataScience/Experimentation.py (hunk 1 above)
Comment (marco-rossi29) [reply DataScience/Experimentation.py]: VW is on my path, and I can run from cmd: `vw --dsjson --cb_adf -d filepath.json.gz`
When I'm running in cmd: `python Experimentation.py ...` it crushes.

File: DataScience/Experimentation.py (hunk 2 above)
Comment (marco-rossi29) [reply DataScience/Experimentation.py]: I'll push a commit with my comment, so you can see if you like it

**Comment by sidsen** [comment]
Sorry, it seems I had changes but didn't update the pull request. I backed them out in favor of yours. Numpy is causing problems in some installations I have -- do you think we should just treat that as a built-in package that everyone will have, or should we avoid depending on it if we don't really need it?

**Pull request merged**